src/commonMain/kotlin/logcat/LogcatLogger.kt

```diff
@@ -0,0 +1,64 @@
+package logcat
+
+/**
+ * Logger that [logcat] delegates to. Call [install] to install a new logger, the default is a
+ * no-op logger. Calling [uninstall] falls back to the default no-op logger.
+ *
+ * You should install [AndroidLogcatLogger] on Android and [PrintLogger] on a JVM.
+ */
+interface LogcatLogger {
+
+  /**
+   * Whether a log with the provided priority should be logged and the corresponding message
+   * providing lambda evaluated. Called by [logcat].
+   */
+  fun isLoggable(priority: LogPriority) = true
+
+  /**
+   * Write a log to its destination. Called by [logcat].
+   */
+  fun log(
+    priority: LogPriority,
+    tag: String,
+    message: String
+  )
+
+  companion object {
+    // For expect functions above
```

Comment:
above => below?

Why do these have to be defined below?

Reply:
Yeah, that should be "below".  They're defined as extensions on the Companion so that they can have the expect modifier.  I would have preferred them inline, but including them in the Companion directly as `expect` complains that there's no backing field or delegate.

> Modifier 'expect' is not applicable to 'member property without backing field or delegate'

I will definitely re-arrange to better co-locate the expected extensions in the file though

Reply:
Rearranged to help with code readability